fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1247,18 +1249,48 @@ public void updateTableMeta(Database db, String tableName, Map<String, String> p
             if (metaValue == olapTable.enableReplicatedStorage()) {
                 return;
             }
+        } else if (metaType == TTabletMetaType.ENABLE_BINLOG) {
+            metaValue = Boolean.parseBoolean(properties.get(PropertyAnalyzer.PROPERTIES_ENABLE_BINLOG));
+            if (metaValue == olapTable.enableBinlog()) {
+                return;
+            }
         } else {
             LOG.warn("meta type: {} does not support", metaType);
             return;
         }
 
         if (metaType == TTabletMetaType.INMEMORY ||
-                metaType == TTabletMetaType.ENABLE_PERSISTENT_INDEX) {
+                metaType == TTabletMetaType.ENABLE_PERSISTENT_INDEX ||
+                metaType == TTabletMetaType.ENABLE_BINLOG ||
+                metaType == TTabletMetaType.BINLOG_TTL ||
+                metaType == TTabletMetaType.BINLOG_MAX_SIZE) {
             for (Partition partition : partitions) {
-                updatePartitionTabletMeta(db, olapTable.getName(), partition.getName(), metaValue, metaType);
+                try {
+                    if (metaType == TTabletMetaType.BINLOG_MAX_SIZE) {
+                        db.readLock();
```

Comment:
Why need a lock here? 
If necessary, it should be put into a `try-finally` block ?